### ISSUE TITLE
ArgoCD overwrites redirectURI param for dex config

### ISSUE THREAD

**Issue by nemo83** [argoproj/argo-cd#1117]
Hi, 

tl;dr

When parsing the dex configuration it seems that the `redirectURI` is overwritten regardless of the fact that it could have been set in the config.

This is preventing me from specifying a different value in the config.

Culprit is:

https://github.com/argoproj/argo-cd/blob/874fe6968323933808926b74a7d5ca0b43d014c5/util/dex/config.go#L59

# Long story

We are running ArgoCD in EKS and we expose it via ELB. The service is (was!) publicly accessible. 
We had to lock it down and restrict access to only trusted ip addresses.

Unfortunately Dex issues a POST on the `/api/dex/callback` using the FQDN name of the ArgoCD server. Because the FQDN is of a public ip address, the call goes outside the AWS thru the public internet (with an AWS ip address) and then hit the ELB. We have to whitelist the AWS regional ip aaddress in order to allow dex to hit the ELB. Ideally dexter should honour the `redirectURI` and issue a local network call to argo w/o going thru the broader internet.

Thoughts?
Gio

**Comment by jessesuen** [comment]
Yes the reason for this is: https://github.com/argoproj/argo-cd/issues/1062#issuecomment-457369217.

We had to do the same workaround that you described (whitelist our own egress IPs). I will duplicate this to issue #1062 to find some way where Argo CD does not have to go out and come back in when it talks to itself as an OIDC client.
